pkg/util/featuregates/featuregates.go

```diff
@@ -0,0 +1,36 @@
+package featuregates
+
+import (
+	v1 "github.com/openshift/api/config/v1"
+	"k8s.io/klog/v2"
+)
+
+const (
+	// DeployMHCControllerFeatureGateName is the name of the feature gate for enabling the MHC controller
+	DeployMHCControllerFeatureGateName = "MachineAPIOperatorDeployMHCController"
+)
+
+// IsDeployMHCControllerEnabled returns if the feature gate for the MHC controller deployment is enabled.
+// For now this is an experimental feature gate, and we only check if it's disabled via the CustomNoUpgrade feature set.
+// The purpose is to disable the MHC controller for being able to test the upcoming MHC feature of the NodeHealthCheck operator.
+// Whenever NHC becomes the default MHC handler, the default return value needs to be changed to false!
+func IsDeployMHCControllerEnabled(fg *v1.FeatureGate) bool {
+	deployMHCControllerFeatureGate := v1.FeatureGateName(DeployMHCControllerFeatureGateName)
+	if fg != nil && fg.Spec.CustomNoUpgrade != nil {
+		for _, enabled := range fg.Spec.CustomNoUpgrade.Enabled {
+			if enabled == deployMHCControllerFeatureGate {
+				klog.V(2).Info("MHC controller enabled by feature gate")
+				return true
+			}
+		}
+		for _, disabled := range fg.Spec.CustomNoUpgrade.Disabled {
+			if disabled == deployMHCControllerFeatureGate {
+				klog.V(2).Info("MHC controller disabled by feature gate")
+				return false
+			}
+		}
+	}
+	// switch to false once NHC is the default!
+	klog.V(4).Info("MHC controller enabled (default)")
+	return true
+}
```

Comment:
The featuregateaccess pattern gives you a way to check if a specific feature gate is enabled or not. We should be using that rather than using our own logic, it helps to co-ordinate between multiple components and versions. I realise we are doing this for testing initially but I suspect at some point we may end up evolving this so we should probably get it right first time.

You can read more [in this FAQ](https://www.google.com/url?sa=t&rct=j&esrc=s&source=appssearch&uact=8&cd=0&cad=rja&q&sig2=JBITYGdVQBjEfwb5Z_p1Mg&ved=0ahUKEwjSlJj414SAAxUE-d8KHe8FBvU4ABABKAAwAA&url=https://drive.google.com/a/redhat.com/open?id%3D1zOL38_KDKwAvsx-LMHfyDdX4-jq3HQU1YWBjfuHaM0Q&usg=AOvVaw3WzByCS-aZkfHZaYqcxCHF)